### PR TITLE
Add missing GRN type for outputs

### LINKED ISSUE #10088: Streams with Outputs cannot be shared
If a stream is configured with a custom output, the share request fails, because we are missing
the `output` type in our GRN registry.

```
2021-02-17T21:59:59.234Z ERROR [AnyExceptionClassMapper] Unhandled exception in REST resource

java.lang.IllegalArgumentException: type <output> does not exist

        at org.graylog.grn.GRNRegistry.lambda$newGRNBuilder$0(GRNRegistry.java:138) ~[graylog.jar:?]

        at java.util.Optional.orElseThrow(Optional.java:408) ~[?:?]

        at org.graylog.grn.GRNRegistry.newGRNBuilder(GRNRegistry.java:138) ~[graylog.jar:?]

        at org.graylog.grn.GRNRegistry.newGRN(GRNRegistry.java:100) ~[graylog.jar:?]

        at org.graylog.security.entities.EntityDependencyResolver.lambda$resolve$1(EntityDependencyResolver.java:70) ~[graylog.jar:?]

        at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195) ~[?:?]

        at java.util.Iterator.forEachRemaining(Iterator.java:133) ~[?:?]

        at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801) ~[?:?]

        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484) ~[?:?]

        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474) ~[?:?]

        at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913) ~[?:?]

        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]

        at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578) ~[?:?]

        at org.graylog.security.entities.EntityDependencyResolver.resolve(EntityDependencyResolver.java:72) ~[graylog.jar:?]

        at org.graylog.security.shares.EntitySharesService.checkMissingPermissionsOnDependencies(EntitySharesService.java:299) ~[graylog.jar:?]

        at org.graylog.security.shares.EntitySharesService.prepareShare(EntitySharesService.java:108) ~[graylog.jar:?]

        at org.graylog.security.rest.EntitySharesResource.prepareShare(EntitySharesResource.java:123) ~[graylog.jar:?]
```

* Graylog Version: any after 4.0

### PR DESCRIPTION
**Pull request by mpfz0r**
Fixes #10088

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog/security/entities/EntityDependencyResolver.java [hunk 1]
```diff
@@ -42,6 +46,11 @@
     private final GRNRegistry grnRegistry;
     private final GRNDescriptorService descriptorService;
     private final DBGrantService grantService;
+    // Some dependencies can be ignored.
+    // E.g. To view a stream with a custom output, a user does not need output permissions
+    private static final Map<GRNType, List<ModelType>> IGNORED_DEPENDENCIES = ImmutableMap.<GRNType, List<ModelType>>builder()
```
Comment (bernd) [inline graylog2-server/src/main/java/org/graylog/security/entities/EntityDependencyResolver.java:51]: How about using a `Set<ModelType>` here? This should make the contains check more efficient and I think we don't need the ordering properties of a list here.

**Review by bernd — CHANGES_REQUESTED** [review]
This fixes the issue for me. Thank you! :+1: 

One one nitpick:

File: graylog2-server/src/main/java/org/graylog/security/entities/EntityDependencyResolver.java (hunk 1 above)
Comment (mpfz0r) [reply graylog2-server/src/main/java/org/graylog/security/entities/EntityDependencyResolver.java]: Yeah, makes sense :+1:

**Review by bernd — APPROVED** [review]
LGTM :+1: Feel free to merge once the build is green.

**Pull request merged**